sqlmesh/core/engine_adapter/mixins.py

```diff
@@ -147,11 +147,19 @@ def _build_table_properties_exp(
                     raise SQLMeshError(
                         f"PARTITIONED BY contains non-column value '{expr.sql(dialect='spark')}'."
                     )
-            properties.append(
-                exp.PartitionedByProperty(
-                    this=exp.Schema(expressions=partitioned_by),
+
+            if self.dialect == "trino" and self.current_catalog_type == "iceberg":
+                # On the Trino Iceberg catalog, the table property is called "partitioning" - not "partitioned_by"
+                # ref: https://trino.io/docs/current/connector/iceberg.html#table-properties
+                properties.append(
+                    exp.Property(this="partitioning", value=exp.Schema(expressions=partitioned_by))
```

Comment:
I'm not super familiar with sqlglot, is this a valid approach?

Reply:
@georgesittas Can you confirm the best approach here?

Reply:
I'm not quite sure - what SQL code is this supposed to generate? Is there a way we can add a test to ensure we generate the correct DDL statement?

One observation is that instead of `this="partitioning"`, we may need to do something like `this=exp.var("partitioning")` to have an expression in `this` vs a string (which is usually the case for `Property` expressions IIRC).

Reply:
Looks good, thanks 👍